frontend/public/components/utils/status-icon.tsx

```diff
@@ -1,37 +1,33 @@
 import * as React from 'react';
 import { Icon } from 'patternfly-react';
 import {CamelCaseWrap} from './camel-case-wrap';
+import * as classNames from 'classnames';
 
-export const StatusIcon: React.FunctionComponent<StatusIconProps> = ({status}) => {
-
-  if (!status){
-    return <span>-</span>;
-  }
-
+export const StatusIcon: React.FunctionComponent<StatusIconProps> = ({status, spin, additionalIconClassName}) => {
   switch (status) {
     case 'New':
-      return <span className="co-icon-and-text"><Icon type="fa" name="hourglass-1" className="co-icon-and-text__icon" /><CamelCaseWrap value={status} /></span>;
+      return <Icon type="fa" name="hourglass-1" className={classNames(spin && 'fa-spin', additionalIconClassName)} />;
```

Comment:
You could pull the `classNames` call up outside the switch so you only need to logic in one place

```tsx
const className = classNames(spin && 'fa-spin', additionalIconClassName);
// ...
className={className}
```

Reply:
Fixed.  Thanks, @spadgett.